modules/core/src/main/scala/doobie/util/fragments.scala

```diff
@@ -17,6 +18,16 @@ object fragments {
   def values[F[_]: Reducible, A](fs: F[A])(implicit w: util.Write[A]): Fragment =
     fr"VALUES" ++ comma(fs.toNonEmptyList.map(f => parentheses(values(f))))
 
+  /** Returns `UPDATE tableName SET columnUpdate0, columnUpdate1, ... WHERE (whereAnd0) AND (whereAnd1) ...`. */
+  def updateSetOpt[F[_]: Foldable](
+      tableName: Fragment,
+      columnUpdates: F[Fragment],
+      whereAnd: F[Fragment]
```

Comment:
Thanks this is a good start!
In my [comment](https://github.com/typelevel/doobie/issues/1893#issuecomment-1632149878) what I was thinking was a fragment for the where condition, so the signature should be `condition: Fragment`

Usage will look like:

```
updateSetOpt(
  fr"my_table",
  ...,
  condition = fr"some_col > 50"
)
```

or a more complicated example
```
updateSetOpt(
  fr"my_table",
  ...,
  condition = fragments.inOpt(fr"name", nameList).getOrElse(fr"false")
)
```

This way the user have full control over the condition.

Reply:
Hi @jatcwang, thanks for the clarification. I updated the signature accordingly.